### PR TITLE
PAPILIO-23-Activity-Creation-UI

### PR DESCRIPTION
**Pull request by Zain-MKhan**
# General info
Implemented static UI for user activity creation

**Issue number**: 
PAPILIO-23

**Task description**: 
 - Static UI for the user to create their own activity

# Testing
To test this feature you have to:

1. Sign in
2. Go to 'My Activities' tab
3. Click on '+ Create Activity'
4. Ensure the layout  matches UI mockups
5. Ensure text fields are editable
6. Tap back arrow at top of page to bring you back to the 'My Activities' tab

# Notes
- The 'IMPORT PICTURE' and '+ CREATE ACTIVITY' buttons found at the bottom of the page are non-functional
- UI tests will be added shortly

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@
 .cxx
 local.properties
 /app/google-services.json
+app/src/main/res/xml/network_security_config.xml
```
Comment (AC159) [inline .gitignore:18]: This file should be committed to version control so we should remove it

File: app/src/main/java/com/soen490chrysalis/papilio/create_activity.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        var actionBar = supportActionBar
+
+        // if Action Bar is not null, then put a back button on it as well as put the "User Profile" title on it
+        if(actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true)
+            actionBar.title = "Create Activity"
+        }
+    }
+    // This is the function that's called when the back button on the action bar is pressed
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
```
Comment (AC159) [inline app/src/main/java/com/soen490chrysalis/papilio/create_activity.kt:22]: Run code formatter with the rules that I posted on discord as the semicolons do not follow the style

File: app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt [hunk 3]
```diff
@@ -38,7 +40,14 @@ class ActivitiesFragment : Fragment()
     ) : View?
     {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_activities, container, false)
+        val view = inflater.inflate(R.layout.fragment_activities, container, false)
```
Comment (AC159) [inline app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt:43]: Use view binding. Here is the documentation on how to do it: [https://developer.android.com/topic/libraries/view-binding#fragments](url)

File: app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt [hunk 4]
```diff
@@ -38,7 +40,14 @@ class ActivitiesFragment : Fragment()
...
+        val createActivityButton = view.findViewById<androidx.appcompat.widget.AppCompatButton>(R.id.activities_create_activity_button)
+
+        createActivityButton?.setOnClickListener {
```
Comment (AC159) [inline app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt:46]: Run code formatter

File: app/src/main/res/layout/activity_create_activity.xml [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+                        <TextView
+                            android:id="@+id/textView10"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content"
+                            android:layout_marginHorizontal="10dp"
+                            android:layout_marginTop="10dp"
+                            android:textColor="@color/black"
+                            android:textSize="20dp"
+                            android:text="Maximum number of participants:" />
+                        <EditText
```
Comment (AC159) [inline app/src/main/res/layout/activity_create_activity.xml:210]: I think the EditTexts can have a more rounded shape to match the UI more closely. I already made such a shape so you can simply add this line: `android:background="@drawable/edit_text_shape"`

File: app/src/main/res/layout/activity_create_activity.xml [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="vertical"
+                    android:layout_marginTop="20dp">
+
+                    <LinearLayout
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content">
+                        <ImageView
+                            android:id="@+id/iv_papilio_logo"
```
Comment (AC159) [inline app/src/main/res/layout/activity_create_activity.xml:39]: The logo should take less space in my opinion

File: app/src/main/res/layout/activity_create_activity.xml [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+                        <TextView
+                            android:id="@+id/textView11"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content"
+                            android:layout_marginHorizontal="10dp"
+                            android:layout_marginTop="10dp"
+                            android:textColor="@color/black"
+                            android:textSize="20dp"
+                            android:text="Import Picture:" />
+                        <Button
```
Comment (AC159) [inline app/src/main/res/layout/activity_create_activity.xml:231]: Button corners should be more rounded

File: app/src/main/res/layout/fragment_activities.xml [hunk 8]
```diff
@@ -18,5 +18,12 @@
             android:layout_height="match_parent"
             android:text="Activities!"
             android:textAlignment="center" />
+
+        <androidx.appcompat.widget.AppCompatButton
```
Comment (AC159) [inline app/src/main/res/layout/fragment_activities.xml:22]: I think we should just use Button here instead of AppCompatButton

**Review by AC159 — CHANGES_REQUESTED** [review]
Couples of things:
- Run code formatter
- Apply shapes to buttons and edit text views to make them match the UI mockups
- Use view binding instead of R.layout in your fragments

File: app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt (hunk 4 above)
Comment (Zain-MKhan) [reply app/src/main/java/com/soen490chrysalis/papilio/view/ActivitiesFragment.kt:46]: Running the formatter on this causes no changes, but when manually putting the open bracket on the next line, android studio throws an error 
image

**Review by AC159 — CHANGES_REQUESTED** [review]
Remove unused imports, incorporate changes from main and we are good to go

**Review by AC159 — APPROVED** [review]
Good to go

**Pull request merged**